File: Go/XTime/time_test.go
// Copyright (c) 2026 FrameworkEase Technologies. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.

package XTime

import (
	"testing"
	"time"

	"github.com/stretchr/testify/assert"
)

func TestCurrent(t *testing.T) {
	assert.WithinDuration(t, Current(), time.Now(), 100*time.Millisecond, "获取的当前时间应在100毫秒误差范围内。")
}

func TestSeconds(t *testing.T) {
	now := time.Now()
	expected := int(now.Sub(Initial()).Seconds())
	seconds := Seconds()
	delta := seconds - expected
	assert.LessOrEqual(t, delta, 1, "获取的秒级时间戳应在1秒误差范围内。")
}

func TestMilliseconds(t *testing.T) {
	now := time.Now()
	expected := int(now.Sub(Initial()).Milliseconds())
	milliseconds := Milliseconds()
	delta := milliseconds - expected
	assert.LessOrEqual(t, delta, 100, "获取的毫秒级时间戳应在100毫秒误差范围内。")
}

func TestMicroseconds(t *testing.T) {
	now := time.Now()
	expected := int(now.Sub(Initial()).Microseconds())
	microseconds := Microseconds()
	delta := microseconds - expected
	assert.LessOrEqual(t, delta, 10000, "获取的微秒级时间戳应在10000微秒误差范围内。")
}

File: CSharp/tests~/Assets/Tests/Runtime/Scripts/XTime/time_test.cs
// Copyright (c) 2026 FrameworkEase Technologies. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.

using System;
using FrameworkEase.Universal.X.Utility;
using NUnit.Framework;

public partial class TestXTime
{
    [Test]
    public void Current()
    {
        var now = DateTime.Now;
        var time = XTime.Current;
        Assert.That((time - now).TotalMilliseconds, Is.LessThanOrEqualTo(100), "获取的当前时间应在100毫秒误差范围内。");
    }

    [Test]
    public void Seconds()
    {
        var now = DateTime.Now;
        var expected = (int)(now - XTime.Initial).TotalSeconds;
        var seconds = XTime.Seconds;
        Assert.That(Math.Abs(seconds - expected), Is.LessThanOrEqualTo(1), "获取的秒级时间戳应在1秒误差范围内。");
    }

    [Test]
    public void Milliseconds()
    {
        var now = DateTime.Now;
        var expected = (long)(now - XTime.Initial).TotalMilliseconds;
        var milliseconds = XTime.Milliseconds;
        Assert.That(Math.Abs(milliseconds - expected), Is.LessThanOrEqualTo(100), "获取的毫秒级时间戳应在100毫秒误差范围内。");
    }

    [Test]
    public void Microseconds()
    {
        var now = DateTime.Now;
        var expected = (now.Ticks - XTime.Initial.Ticks) / 10;
        var microseconds = XTime.Microseconds;
        Assert.That(Math.Abs(microseconds - expected), Is.LessThanOrEqualTo(10000), "获取的微秒级时间戳应在10000微秒误差范围内。");
    }
}
